src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs

```diff
@@ -112,6 +112,14 @@ private string OsVersion
         /// <returns>Value for provided HKLM key</returns>
         private string ReadHKLMRegistry(string path, string key)
         {
+            switch (Environment.OSVersion.Platform)
```

Comment:
@dsgouda let's test this on mono/non-windows to make sure this does not throw runtime exceptions.